demisto_sdk/commands/content_graph/parsers/integration_script.py

```diff
@@ -3,29 +3,40 @@
 from pathlib import Path
 from typing import Any, Dict, List, Optional, Set
 
+import pydantic
+
 from demisto_sdk.commands.common.constants import MarketplaceVersions
 from demisto_sdk.commands.common.docker.docker_image import DockerImage
 from demisto_sdk.commands.common.tools import get_value
 from demisto_sdk.commands.content_graph.common import ContentType, RelationshipType
 from demisto_sdk.commands.content_graph.parsers.yaml_content_item import (
     YAMLContentItemParser,
 )
+from demisto_sdk.commands.content_graph.strict_objects.script import StrictScript
 from demisto_sdk.commands.prepare_content.integration_script_unifier import (
     IntegrationScriptUnifier,
 )
 
 
 class IntegrationScriptParser(YAMLContentItemParser):
     def __init__(
-        self,
-        path: Path,
-        pack_marketplaces: List[MarketplaceVersions],
-        git_sha: Optional[str] = None,
+            self,
+            path: Path,
+            pack_marketplaces: List[MarketplaceVersions],
+            git_sha: Optional[str] = None,
     ) -> None:
         self.is_unified = YAMLContentItemParser.is_unified_file(path)
         super().__init__(path, pack_marketplaces, git_sha=git_sha)
         self.script_info: Dict[str, Any] = self.yml_data.get("script", {})
         self.connect_to_api_modules()
+        self.structure_errors = self.validate_structure()
+
+    def validate_structure(self) -> Optional[list[dict]]:
+        try:
+            StrictScript(**self.yml_data)
+        except pydantic.error_wrappers.ValidationError as e:
+            return e.errors()
+        return None
```

Comment:
let's move this code to a higher level of class from which all parsers inherit, and store the specific strict model somewhere in each inheriting class (e.g. `ScriptParser` will hold some attribute/key to some mapping, that will allow it to know it should use `StrictScript`, and same for each parser and strict class)